src/wms/src/main/java/org/geoserver/wms/featureinfo/HTMLFeatureInfoOutputFormat.java

```diff
@@ -68,7 +68,7 @@ public TemplateModel wrap(Object object) throws TemplateModelException {
         });
     }
 
-    GeoServerTemplateLoader templateLoader;
+    volatile GeoServerTemplateLoader templateLoader;
```

Comment:
Yes, you are correct.  The volatile modifier is not necessary now that it is in the synchronized block.  I will remove it from the pull request.